### ISSUE TITLE
{ERR} Ro@http://localhost:8000/W:5:5208

### ISSUE THREAD

**Issue by 89z** [ZEDCWT/ZED.CWT.CrabSave#3]
If I put a URL in the browser, I get this result:

~~~
`https://www.youtube.com/watch?v=mvDWtMVPyf8`
YouTube Video mvDWtMVPyf8
{ERR} Ro@http://localhost:8000/W:5:5208
lD/g[EI]@http://localhost:8000/W:5:148495
~~~

**Comment by ZEDCWT** [comment]
Are you able to access the URL directly? If you are behind a proxy, did you configure it in the setting page?
This error should contain readable message like this
```
`https://www.youtube.com/watch?v=mvDWtMVPyf8`
YouTube Video mvDWtMVPyf8
{ERR} Error: Status not satisfied | 0
    at XMLHttpRequest.g.(anonymous function) (http://localhost/xxx/W:5:148495)
```
indicating that the HTTP status code is not expected.
So, which browser are you using? I may also need to change the way to print errors

**Comment by 89z** [comment]
> Are you able to access the URL directly?

Yes

> If you are behind a proxy, did you configure it in the setting page?

No proxy, just normal connection

> So, which browser are you using?

Firefox

**Comment by ZEDCWT** [comment]
Okay, did not notice that not all browsers include error messages in the stacks.

Now, are you able to access `https://www.googleapis.com/`? It should show a 404 page
If so, please run the following code in the devtools opened from the Save page
```
Wish.B.ReqB('Api/~/' + Wish.C.UE('https://www.googleapis.com/youtube/v3/videos?key=#GoogleAPIKey#&part=id,snippet,contentDetails&id=mvDWtMVPyf8')).Now(B => console.log('Loaded',B),E => console.log('Error',E))
```
And then let me see the output

**Comment by 89z** [comment]
> Now, are you able to access `https://www.googleapis.com/`? It should show a 404 page

I dont understand the point of this question. If it doesnt work for you, why would it work for me? and if it doesnt work for you or me, then why even ask the question?

> If so, please run the following code in the devtools opened from the Save page

I dont know what the "Save page" is.

**Comment by ZEDCWT** [comment]
Well, it is the typical 'it works on my machine' situation, and the point here is to perform connection test since loading basic video information relies on Google API service. So the first step is to figure out from which layer does the service not accessible anymore.

I mean to open devtools while visiting `http://localhost:8000/`, since the test code snippet requires components loaded on it.

**Comment by 89z** [comment]
> Well, it is the typical 'it works on my machine' situation, and the point here is to perform connection test since loading basic video information relies on Google API service.

No its not, because, by your own admission, `https://www.googleapis.com/`, **doesnt work on your machine**. So again, what is the point of that test?

**Comment by ZEDCWT** [comment]
It works
What make you falsely think it does not work on my machine?

**Comment by 89z** [comment]
You said here [1]:

> Now, are you able to access `https://www.googleapis.com/`? It should show a 404 page

So, I guess I dont understand what you were saying. Are you saying that it should show a 404 page for you and me, or just me? And if just me, what makes you think that the page would work for you and not me? Also, I seriously doubt it even works for you. You should run `curl -I https://www.googleapis.com/` and see what you get.

1. https://github.com/ZEDCWT/ZED.CWT.CrabSave/issues/3#issuecomment-883513119

**Comment by ZEDCWT** [comment]
This IS the expected behaviour.
Now I get your point, we have different definition of 'work', so we'd better stop using such ambiguous words from now on.

Let me rewrite the logic.
Fist test is `GET{path = /} => HTTP404`, as said earlier, it is a connection test, so `being able to see the 404 page` means there are no connection issues, also it does NOT imply `GET{path != /} => HTTP404`.
Though you did not explicitly answer the question, I assume you also see the 404 page, so I guess there are no connectivity issue on your side.
Then, the next test is to manually request API data though CrabSave, and the exact output is required to perform further investigation.
There was a typo of the code snippet above, the corrected version is
```
Wish.B.ReqB('Api/~' + Wish.C.UE('https://www.googleapis.com/youtube/v3/videos?key=#GoogleAPIKey#&part=id,snippet,contentDetails&id=mvDWtMVPyf8')).Now(B => console.log('Loaded',B),E => console.log('Error',E))
```

**Comment by 89z** [comment]
Here is why your test is pointless. Even if I ping a made up page, I get your same "success" value:

~~~
PS C:\> curl -I https://www.googleapis.com/z
HTTP/2 404
~~~

You need to either define what the HTML response should be (empty, `<!DOCTYPE>`, etc), or actually pick a path on that domain where the "success" value is not 404.

**Comment by 89z** [comment]
Closing this, because I dont care anymore

**Comment by ZEDCWT** [comment]
![Pony](https://user-images.githubusercontent.com/7867810/126732320-309f4111-fbf9-412c-9a1d-5b6d0c48b152.png)
> You need to either define what the HTML response should be (empty, <!DOCTYPE>, etc), or actually pick a path on that domain where the "success" value is not 404.

I'd say, the problem here is that you have trouble reading, also not trying to be corporative at all.
 From my last message, i already said that the first test is CONCLUDED with `I guess there are no connectivity issue on your side`.
So you are suppose to continue on the next test rather than struggling on some nonsense 'made up' 404 test.

> actually pick a path on that domain where the "success" value is not 404
This is EXACTLY what the next test is designed for
![Pony](https://user-images.githubusercontent.com/7867810/126732703-50ba476d-a065-4f85-bb6e-8717b68339a7.png)
Can't you read it? Isn't is what you are looking for? A non 404 path?

**Comment by 89z** [comment]
Just tell me what you want me to do, in plain english. And dont have me testing any URLs that are 404, as thats pointless. If you have a sane, legitimate test that makes sense, I will do it. Otherwise we are done here. I would like to help you troubleshoot the problem, but youre making this way harder than it needs to be. Dont have people testing known bogus URLs.

**Comment by ZEDCWT** [comment]
![A](https://user-images.githubusercontent.com/7867810/126748553-17640a0f-2cc6-43a1-8757-a311926f311a.png)

![B](https://user-images.githubusercontent.com/7867810/126748556-114e940c-ed74-45d9-a3e0-6f854295369c.png)

![C](https://user-images.githubusercontent.com/7867810/126748560-c7f5b27a-5f79-40e8-91f0-7acd54aceb59.png)

I already posted three times, why pretend to not see? Why keep letting me repeat my words?
Plain English, okay
+ Copy the provided code snippet
+ Start CrabSave and open http://localhost:8000 using FireFox
+ Open DevTools by F12, and navigate to the Console tab
+ Paste the code snippet, and tap Enter key to execute it
+ Wait till it shows `Error` or `Loaded`
+ Take a scrrenshot by tapping `PrintScreen` key or using other methods
+ And drag the screenshot to the comment textarea of this issue page
+ Click `Comment` button to post

**Comment by 89z** [comment]
~~~
Uncaught ReferenceError: Wish is not defined
~~~

**Comment by ZEDCWT** [comment]
![A](https://user-images.githubusercontent.com/7867810/126855859-8025cb4f-d77f-4d74-bec5-1d5f829db2bc.png)
Did you run the code on the devtools opened from the CrabSave page

**Comment by 89z** [comment]
You said go to http://localhost:8000 - so thats where I went

**Comment by ZEDCWT** [comment]
I also said `Start CrabSave`. Did you started on a different port?

**Comment by 89z** [comment]
My point is, you said go to http://localhost:8000, but your screenshot shows http://localhost/Save/

You are unreliable and inconsistent at giving instruction.

**Comment by ZEDCWT** [comment]
This screenshot is showing that the devtools is opened from the CrabSave page.
So you did not started CrabSave.
Forget about it. You seem not to be capable to follow such instructions.
Let us make it simpler.
+ Download Google Chrome
+ Open Google Chrome
+ Repeat the steps of your first comment but using Google Chrome instead of FireFox this time
+ Paste the result back.

**Comment by 89z** [comment]
You said:

> Start CrabSave and open http://localhost:8000 using FireFox

which I did. but then you screenshot shows `http://localhost/Save/`. You never say how to get to that address, you just assume I know how to do it, even though I have never used the program before.

**Comment by ZEDCWT** [comment]
Again, you start to focus on unnecessary details and ignoring my words.
> ![Pony](https://user-images.githubusercontent.com/7867810/127817814-06c05ce5-3fec-423b-877a-7f9122421122.png)

Why you believe the URL IS the one make things different?

>  even though I have never used the program before.

Yes you had.
![Pony](https://user-images.githubusercontent.com/7867810/127818000-9e992b4f-df6e-4add-b89f-26b07ae31e29.png)
You see, you used this program before and successfully opened `http://localhost:8000/`. So why the heck do i have to teach you how to get to that address when you are absolutely able to do it?

> ![Pony](https://user-images.githubusercontent.com/7867810/127818373-cbe4d264-719a-44e6-a6c2-be3555c1af66.png)

Again, I'm forced to repeat my words.
I already provided an alternate method, why not follow this instead? Are you brain designed to be single-threaded or mentally handicapped?

**Comment by 89z** [comment]
You say go to http://localhost:8000, but screenshot shows http://localhost/Save/. So you need to either correct your instruction, or correct you screenshot. End of story.

**Comment by ZEDCWT** [comment]
Why you are not able to listen????
Why you are still struggling on somethnig you have trouble understanding????
Why you keep making me repeating the same thing????
I said already. This screenshot is SHOWING THAT THE DEVTOOLS IS OPENED FROM THE CRABSAVE PAGE.
Do you understand what the meaning of `REFERENCE PICTURE` is???
And there are ABSOLUTELY NOTHING related to THE STUPID URL
The instruction is CORRECT and the screenshot is ALSO CORRECT

Why dont you use you BRAIN before commenting? Or you dont have one??

You want the 8000. Okay, let me feed this to you
![Pony](https://user-images.githubusercontent.com/7867810/128804626-736d878e-d295-4215-9aca-1f63cfbeec78.png)
Does it change anything? No, it still shows that you are the retarded bitch.
